src/HDF5.jl

```diff
@@ -710,6 +710,12 @@ function h5read(filename, name::String, indices::Tuple{Vararg{Union{Range{Int},I
     dat
 end
 
+function h5read{N}(filename, name::String, cr::CartesianRange{CartesianIndex{N}})
+    # transfer to unit range
+    ur = map((x,y)->x:y, cr.start.I, cr.stop.I)
```

Comment:
Better to use `first(cr)` and `last(cr)` than `cr.start.I` and `cr.stop.I`; in 0.7, the representation of CartesianRange has [changed](https://github.com/JuliaLang/julia/pull/20974). Indeed, in 0.7 this can just be `ur = cr.indices`.